### DISCUSSION TITLE
历史版本在那里

### DISCUSSION THREAD

**Discussion by gametta** [ophub/amlogic-s9xxx-armbian#1421] [Other]
最近使用新版本遇到一些问题，以前用老版本没有这个问题，因为总是喜欢更新，造成老版本找不到的问题，
麻烦O大能够提供一下历史版本下载的地方，

**Comment by ophub** [comment]
没有历史版本，周更周删。

**Comment by gametta** [comment]
这个习惯不好啊，最起码保持半年再删除，如果没有空间我可以保存啊
或者说怎么能快速下载全部文件。

**Comment by ophub** [comment]
编译一次，生成的 releases 是 40G，一周编译一次最新版，1个 `Jammy` 加1个 `Bullseye` 共 `80G` 存储空间。
我觉得 github.com 免费提供的服务挺好，咱别给人浪费。有新编译的就及时删除旧版，给他们节约点服务器资源。

> 今日 TUNA 一台后端服务器存储占用率来到了 99.8%（剩余 177G，总空间 90T）
https://github.com/tuna/issues/issues/1737#issuecomment-1543555024

刚才我去清华源申请加入镜像，但是清华目前服务器满了，短期内没有新增镜像的计划。等以后有机会了我再继续申请。

**Comment by gametta** [comment]
O大的工作量很大，我看到了，我也希望能够尽一份力，我可以提供一个ftp，目前剩余空间30T左右，动态域名，300M上，如果许可的话，可以再分发到其他网盘分流。

**Comment by ophub** [comment]
好啊，非常感谢你的支持。

https://mirrors.tuna.tsinghua.edu.cn/armbian-releases/
你会做这种镜像站的存储吗？或者类似这种的文件管理器方式，能提供目录浏览和文件下载的。
放 Armbian 和 kernel 

https://github.com/ophub/amlogic-s9xxx-armbian/releases/tag/Armbian_jammy_05.09.2000
https://github.com/ophub/kernel/releases/tag/kernel_stable

镜像有2种: Jammy 和 Bullseye
内核有4中: stable, flippy, rk3588, dev

**Comment by gametta** [comment]
这种应该是一个http服务器上面的目录文件浏览，安全方面有待考虑，我这边先测试一下，我更倾向于ftp按照日期或者版本直接放里面，管理方便。

**Comment by ophub** [comment]
可以。文件支持wget下载即可。

**Comment by JFLim1** [comment]
> There are 4 kernels: stable, flippy, rk3588, dev

Hi @ophub,

For normal daily use on Beelink GT King Pro should I use kernel-stable or kernel-flippy?  What is the difference between kernel-stable and kernel-flippy?  For example I notice there are kernel-6.1.27-ophub and kernel-6.1.27-flippy-83+ available.

**Comment by Scirese** [comment]
其实我们可以减少一下生成镜像的数量, 目前我们要为每个 SoC 都生成一份镜像, 而这些镜像的差别仅仅就只是启动配置和 motd. 显然没必要仅仅为了这一点差别就重新生成一份镜像, 把能统一的都统一起来, 每次 release 只需要发4份镜像 - Amlogic, Allwinner, Rockchip 和 Rock5B, 都按 800M 计算, 总大小还不到 3.5G. 
如何修改启动配置在文档里有讲, 这个步骤交给用户是完全可以的. 镜像还可以使用 xz 来最大程度的压缩, 算下来这个使用量完全可以在 Github 上保留几个版本.

稍久远的历史版本可以发到 Onedrive 上, Onedrive 很轻松就能拿到 5T 免费空间, 再搭配 Alist 这些, 也完全能做到镜像站的效果.

**Comment by ophub** [comment]
> For normal daily use on Beelink GT King Pro should I use kernel-stable or kernel-flippy? What is the difference between kernel-stable and kernel-flippy? For example I notice there are kernel-6.1.27-ophub and kernel-6.1.27-flippy-83+ available.

The [kernel_flippy](https://github.com/ophub/kernel/releases/tag/kernel_flippy), [kernel_stable](https://github.com/ophub/kernel/releases/tag/kernel_stable), and [kernel_dev](https://github.com/ophub/kernel/releases/tag/kernel_dev) are universal kernels that can be interchanged. Flippy's feature is being more user-friendly, Stable is more intimate, and Dev is smoother.

look at: https://github.com/ophub/kernel

**Comment by ophub** [comment]
https://github.com/armbian/community/releases/tag/23.05.0-trunk-e33842ef2

我看了下armbian的releases里，一个tags下保存了3千多个文件。我想我是不是也狠心一点多保存一些吧。
暂定以后的tags按月创建，每个月的历史镜像都增量保存不删档了。

**Comment by JFLim1** [comment]
> The [kernel_flippy](https://github.com/ophub/kernel/releases/tag/kernel_flippy), [kernel_stable](https://github.com/ophub/kernel/releases/tag/kernel_stable), and [kernel_dev](https://github.com/ophub/kernel/releases/tag/kernel_dev) are universal kernels that can be interchanged. Flippy's feature is being more user-friendly, Stable is more intimate, and Dev is smoother.

Hi @ophub,

Thanks for the response.  Based on the kernel size taking kernel-6.1.28 as example, flippy-72.1MB, Stable-85.8MB and Dev-90.1MB, is it safe to assume that Stable has additional patches over Flippy and Dev has additional patches over Stable?

**Comment by ophub** [comment]
Small differences between the 3 kernel series:

`Flippy`: This is the standard, original version. It is characterized by being more user-friendly.

`Stable`: `More support options` have been gradually enabled based on user requests in the Flippy config template over the years. It is characterized by being more thoughtful.

`Dev`: The config used is the same as stable, but when compiling the kernel source locally, several `patches` provided by users were added. For example, the `TEXT_OFFSET` patch was added to the 5.10/6.1 series, allowing this series of kernels to be used on devices without a dedicated uboot. It is characterized by being smoother.

Using an iPhone analogy: Flippy is iPhone X, Stable is iPhone X Pro, and Dev is iPhone X Max. However, `the incremental changes in Stable and Dev are not universally necessary`. Since the Stable and Dev series are incremental changes, they can theoretically be interchanged with the Flippy series. However, because I don't understand the kernel, I'm not sure if these increments have negative effects. Therefore, `it is recommended to use the Flippy series on main devices` and switch around with other experimental devices using the Stable and Dev series.

Because the Stable series enables more options than the Flippy series and solves many functional support problems historically raised in problematic areas, `these enabled options are generally applicable`. Therefore, starting this month, `I have made Stable the default kernel`.

Previously, when someone had a new extension option requirement, I would ask Flippy to help choose these options by default in future configs. However, he has been deeply researching several models in the past two months, and it's not convenient for me to disturb him every day. Additionally, these options, which are the same as what you see, have significantly increased the size of the kernel package, which may not be the best approach. After thinking it over, I decided to establish an extended kernel series specifically tailored to Armbian requirements. So now, Stable will independently use a different config template for online compilation and is stored in the kernel repository. If there are additional expansion requirements, users can also customize and add more options.

3个系列内核的小区别：

flippy: 这是标准，原版转存的。特点是更懂你。

stable: 根据用户要求，在flippy的config模板上逐年累月地启用了更多支持选项。特点是更贴心。

dev: 使用的config与stable相同，但内核源码在本地编译时，多打了几个用户提供的补丁，例如5.10/6.1也打了TEXT_OFFSET补丁，可以在没专用uboot的设备里使用这个系列的内核了。特点是更顺滑。

拿IPHONE类比一下：flippy 是 iPhone X，stable是iPhone X Pro，Dev是iPhone X Max。但是stable和dev的增量变化并不是普遍需要的。由于stable和dev系列是增量变化，所以理论上是可以和flippy系列互换使用。但由于我不了解内核，不确定这些增量是否会产生负面影响。因此，建议在主要设备上使用Flippy系列，在其他实验设备上使用Stable和Dev系列。

由于Stable系列比Flippy系列启用了更多选项，解决了历史上出现在问题区域的一些功能不支持的问题，而且这些启用的选项具有普遍适用性。因此，从本月开始，我已将Stable作为默认内核。

以前，如果有人有新的扩展选项需求，我会请Flippy大佬在未来的配置文件中默认选择这些选项。最近这2个月他比较忙，在研究适配几个新设备，我总是打扰他也不好。而且，这些选项和你看到的结果一样，显著增加了内核软件包的大小，这可能不是最好的方法，因为这些选项可能对openwrt来说是无用的。因此，现在Stable将独立使用不同的配置模板进行在线编译，并存储在内核仓库中。如果有更多的扩展需求，用户也可以自定义添加更多选项。

**Comment by JFLim1** [comment]
Hi @ophub,

Thanks for the detailed response.
